server/handlers/validation.go

```diff
@@ -0,0 +1,377 @@
+package handlers
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/endophage/gotuf"
+	"github.com/endophage/gotuf/data"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/notary/server/storage"
+	"github.com/endophage/gotuf/keys"
+	"github.com/endophage/gotuf/signed"
+	"github.com/endophage/gotuf/utils"
+)
+
+// ErrValidation represents a general validation error
+type ErrValidation struct {
+	msg string
+}
+
+func (err ErrValidation) Error() string {
+	return fmt.Sprintf("An error occurred during validation: %s", err.msg)
+}
+
+// ErrBadHeirarchy represents a missing snapshot at this current time.
+// When delegations are implemented it will also represent a missing
+// delegation parent
+type ErrBadHeirarchy struct {
+	msg string
+}
+
+func (err ErrBadHeirarchy) Error() string {
+	return fmt.Sprintf("Heirarchy of updates in incorrect: %s", err.msg)
+}
+
+// ErrBadRoot represents a failure validating the root
+type ErrBadRoot struct {
+	msg string
+}
+
+func (err ErrBadRoot) Error() string {
+	return fmt.Sprintf("The root being updated is invalid: %s", err.msg)
```

Comment:
Might be more user-friendly to say "The root metadata file" or something like that.